### PR TITLE
Hr Attendence Module

### PR DESCRIPTION
**Pull request by rorpyll**

### CODE REVIEW

File: app/controllers/hr/attendences_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        else
+          render :edit
+        end
+      else
+        redirect_to hr_attendence_path
+      end
+    end
+
+    def search
+      @attendences = if params[:start_date].blank? && params[:end_date].blank?
```
Comment (salvimohan) [inline app/controllers/hr/attendences_controller.rb:32]: move logic in Model

File: app/views/hr/attendences/_attendence.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<% array = [] %>
+    <% @attendences.each do |attendence| %>
+                   <%# if current_employee.id == attendence.employee_id %>
+        <% unless array.include?(attendence.created_at.strftime('%d-%m-%y')) %>
+            <% array << attendence.created_at.strftime('%d-%m-%y') %>
+            <tr>
```
Comment (salvimohan) [inline app/views/hr/attendences/_attendence.html.erb:6]: you can use same partial before you have created

File: spec/controllers/hr/application_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+RSpec.describe Hr::ApplicationController, type: :controller do
+  fixtures :all
+
+  let!(:designation) { FactoryBot.create(:designation) }
+  let!(:hr) { FactoryBot.create(:employee, role_id: roles(:role_one).id, designation_id: designation.id) }
+  let!(:employee) { FactoryBot.create(:employee, role_id: roles(:role_two).id, designation_id: designation.id) }
+  let!(:attendence) { FactoryBot.create(:attendence, employee_id: employee.id) }
+
+  describe 'Only all for Hr' do
```
Comment (rorshaan) [inline spec/controllers/hr/application_controller_spec.rb:13]: typo mistake

File: spec/controllers/hr/application_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    it 'response successfully when current employee role is hr' do
+      sign_in(hr)
+      controller.send(:authorize_admin)
+      expect(response.status).to eq(200)
+    end
+
+    it 'return root path when current employee role is Employee' do
+      sign_in(employee)
+      employee.update_column('role_id', roles(:role_two).id)
+      expect(response.status).to eq(200)
```
Comment (rorshaan) [inline spec/controllers/hr/application_controller_spec.rb:23]: specs is not clear.

Comment (rorshaan) [inline spec/controllers/hr/application_controller_spec.rb:23]: status will not be 200(success)

File: spec/controllers/hr/attendence_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  let!(:attendence) { FactoryBot.create(:attendence, employee_id: employee.id) }
+
+  before do
+    sign_in(employee)
+  end
+
+  describe '#show attendences' do
+    it 'Hr can see all employee attendences' do
+      get :show_attendence, params: { id: employee.id }
+      expect(response.status).to eq(200)
```
Comment (rorshaan) [inline spec/controllers/hr/attendence_controller_spec.rb:19]: Expect with instance variable value

File: app/controllers/hr/attendences_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+module Hr
+  class AttendencesController < ApplicationController
+    before_action :set_attendance, only: %i[show edit update]
+
+    def show_attendence
+      @employee = Employee.find(params[:id])
+      @attendences = Attendence.where(employee_id: @employee.id)
+    end
+
+    def show
```
Comment (rorshaan) [inline app/controllers/hr/attendences_controller.rb:13]: If possible dry method by using common mettod for show and show_attendence

**Review by rorshaan — CHANGES_REQUESTED** [review]

**Review by rorshaan — APPROVED** [review]

**Pull request merged**